SwiftCompilerSources/Sources/SIL/KeyPath.swift

```diff
@@ -0,0 +1,285 @@
+//===--- KeyPath.swift ----------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2025 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import AST
+import SILBridging
+
+//============================================================================//
+// KeyPathPattern
+//============================================================================//
+
+public struct KeyPathPattern {
+  public let bridged: BridgedKeyPathPattern
+
+  public init(bridged: BridgedKeyPathPattern) {
+    self.bridged = bridged
+  }
+
+  public init(
+    _ function: Function,
```

Comment:
Passing the `function` is a bit confusing, because the pattern will not reference the function. It's actually only used to get a handle to the SILModule.
It might be better to move this initializer to SILBuilder (where all the other SIL creation APIs live). The builder has access to the SIL module.

And again: please use argument labels.